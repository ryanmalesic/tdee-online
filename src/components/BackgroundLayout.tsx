import React from 'react';

import styles from './BackgroundLayout.module.scss';

const BackgroundLayout: React.FC = (props) => {
  const { children } = props;

  return (
    <section className={`hero is-fullheight ${styles.background}`}>
      <div className="hero-body">
        {children}

        <span className={`tag ${styles.credit}`}>
          Photo by &nbsp;
          <a href="https://unsplash.com/@brett_jordan?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">
            Brett Jordan
          </a>
          &nbsp;on&nbsp;
          <a href="https://unsplash.com/?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">
            Unsplash
          </a>
        </span>
      </div>
    </section>
  );
};

export default BackgroundLayout;
