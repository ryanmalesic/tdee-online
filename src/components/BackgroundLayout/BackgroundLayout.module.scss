@import '../../scss/theme.scss';

.background {
  background-image: url('../../../public/background.jpg');
  background-position: center;
  background-size: cover;
}

.heroHead {
  background-color: $background;
}

.footer {
  padding-bottom: 1rem;
  padding-right: 1rem;
  text-align: right;
}
