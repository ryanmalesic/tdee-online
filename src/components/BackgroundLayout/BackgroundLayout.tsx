import React from 'react';

import Navbar from '../Navbar';
import styles from './BackgroundLayout.module.scss';

const BackgroundLayout: React.FC = (props) => {
  const { children } = props;

  return (
    <section className={`hero is-fluid is-fullheight ${styles.background}`}>
      <div className={`hero-head ${styles.heroHead}`}>
        <Navbar />
      </div>
      <div className="hero-body">{children}</div>
      <div className={`hero-footer ${styles.footer}`}>
        <span className="tag">
          Photo by &nbsp;
          <a href="https://unsplash.com/@brett_jordan?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">
            Brett Jordan
          </a>
          &nbsp;on&nbsp;
          <a href="https://unsplash.com/?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">
            Unsplash
          </a>
        </span>
      </div>
    </section>
  );
};

export default BackgroundLayout;
