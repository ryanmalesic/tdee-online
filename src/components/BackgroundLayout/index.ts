import BackgorundLayout from './BackgroundLayout';

export { BackgorundLayout as default };
