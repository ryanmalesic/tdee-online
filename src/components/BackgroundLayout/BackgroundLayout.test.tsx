import React from 'react';
import renderer from 'react-test-renderer';

import BackgroundLayout from './BackgroundLayout';

it('renders BackgroundLayout unchanged', () => {
  const tree = renderer.create(<BackgroundLayout />).toJSON();
  expect(tree).toMatchSnapshot();
});
