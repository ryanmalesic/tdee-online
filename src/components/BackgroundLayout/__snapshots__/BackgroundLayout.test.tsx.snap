// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BackgroundLayout unchanged 1`] = `
<section
  className="hero is-fullheight background"
>
  <div
    className="hero-body"
  >
    <span
      className="tag credit"
    >
      Photo by  
      <a
        href="https://unsplash.com/@brett_jordan?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
      >
        Brett Jordan
      </a>
       on 
      <a
        href="https://unsplash.com/?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
      >
        Unsplash
      </a>
    </span>
  </div>
</section>
`;
